RiotSwiftUI/Modules/Spaces/AddRoomSelector/Service/AddRoomItemsProcessor.swift

```diff
@@ -0,0 +1,69 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+class AddRoomItemsProcessor: MatrixItemChooserProcessorProtocol {
+    
+    // MARK: Private
+    
+    private let parentSpace: MXSpace
+    
+    // MARK: Setup
+    
+    init(parentSpace: MXSpace) {
+        self.parentSpace = parentSpace
+    }
+    
+    // MARK: MatrixItemChooserSelectionProcessorProtocol
+    
+    var dataType: MatrixItemChooserType {
+        .room
+    }
+
+    func computeSelection(withIds itemsIds: [String], completion: @escaping (Result<Void, Error>) -> Void) {
+        addChild(from: itemsIds, at: 0, completion: completion)
+    }
+    
+    func isItemIncluded(_ item: (MatrixListItemData)) -> Bool {
+        return !parentSpace.isRoomAChild(roomId: item.id)
+    }
+    
+    // MARK: Private
+    
+    func addChild(from roomIds: [String], at index: Int, completion: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
It would be nice to have a doc comment on this one as it is named `addChild` but then recursively adds all of them. Took me a while to understand what this actually does.